docs/using-the-jdbc-driver/using-plugins/UsingTheAuroraInitialConnectionStrategyPlugin.md

```diff
@@ -0,0 +1,31 @@
+# Aurora Initial Connection Strategy Plugin
+The AWS Advanced JDBC Driver allows the user to configure their initial connection strategy using this plugin. With this plugin enabled, the initial connection will be chosen based on the configured strategy. The time between retries and timeouts for these initial connections can also be configured.
+<br><br>
+When a user uses cluster endpoints, then the actual instance for a new connection is resolved by DNS. During failover, the cluster elects another instance. DNS is updated, but it takes time (up to 40–60s). While DNS is updating, if a user tries to connect to the endpoint, it gets connected to an old node. This plugin is used to replace the old endpoint with a current endpoint while DNS is still updating. Changing the initial connection strategy would change which endpoint is chosen to replace the old one in this scenario.
```

Comment:
```suggestion
This plugin is used to replace the old endpoint with a current endpoint while DNS is still updating. When a user connects to a cluster endpoint, the actual instance for a new connection is resolved by DNS. During failover, the cluster elects another instance to be the writer. While DNS is updating, which can take up to 40-60 seconds, if a user tries to connect to the endpoint they may be connected to an old node.  Changing the initial connection strategy would change which endpoint is chosen to replace the old one in this scenario.
```

Reply:
When changing the initial connection strategy is mentioned, is this for reader cluster endpoints?

Reply:
Can part of this section be combined with the first paragraph so that everything this plugin does is mentioned at the start?